src/bench/pqs/spraylist.cpp

```diff
@@ -55,14 +55,38 @@ void
 spraylist::insert(const uint32_t &k,
                   const uint32_t &v)
 {
-    sl_add(m_q, k, READ_ADD_REM_ELASTIC_TX);
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
+}
+
+void
+spraylist::insert(const size_t &k,
+                  const size_t &v)
+{
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
 }
 
 bool
-spraylist::delete_min(uint32_t &v)
+spraylist::delete_min(size_t &k, size_t &v)
 {
-    const int ret = spray_delete_min(m_q, &v, d);
-    return (ret == 1);
+    unsigned long k_ret;
+    unsigned long v_ret;
+    int ret;
+    do {
+        ret = spray_delete_min_key(m_q, &k_ret, &v_ret, d);
```

Comment:
I did it this way because it makes the file_shortest_paths benchmark (src/bench/file_shortest_paths.cpp) together with the SprayList behave in the same way as the shortest paths benchmark implemented by @jkopinsky (lib/spraylist_linden/sssp.c). I don't see any other good way to accomplish this but I'm open for suggestions.

If I'm not mistaken, the file_shortest_paths benchmark used with the SprayList could compute the wrong result if this loop was not present even though it is probably very unlikely in practice. The reason is that the spurious failures seems to depend on a random number generator so if all worker threads spuriously fail enough times at the wrong time the algorithm will stop even though there is still work to do. Again if I'm not mistaken, the file_shortest_paths benchmark together with KLSM will always compute the correct result even though KLSM can spuriously fail. The reason is that all worker threads can't fail "at the same time" with KSLM. The stop condition for the file_shortest_paths benchmark is that all worker threads called delete_min after some time point after which no insert operations have been issued and got false as return value.